### PR TITLE
Tornado

### PR DESCRIPTION
**Pull request by aokellermann**
### Changes
* Separates client and server
* Implements network communication with SSL
* Removes some unneeded components (such as Command)

### CODE REVIEW

**Review by BobbyBytes — DISMISSED** [review]
Looks good to me. 
Port 6969..... Nice.

File: bin/securedrop [hunk 1]
```diff
@@ -1,7 +1,8 @@
 #!/usr/bin/env python
 
 import os
-from securedrop.client import Client
+
```
Comment (BobbyBytes) [inline bin/securedrop:4]: Is this an erroneous extra line?

**Review by jjjoycing123 — APPROVED** [review]

File: securedrop/server.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+from Crypto.Cipher import AES
+import Crypto.Util.Padding
+
+from securedrop import ServerBase
+from securedrop.register_packets import REGISTER_PACKETS_NAME, RegisterPackets
+from securedrop.status_packets import STATUS_PACKETS_NAME, StatusPackets
+from securedrop.login_packets import LOGIN_PACKETS_NAME, LoginPackets
+from securedrop.add_contact_packets import ADD_CONTACT_PACKETS_NAME, AddContactPackets
+
+sd_filename = 'server.json'
```
Comment (BobbyBytes) [inline securedrop/server.py]: CONST sd_filename should be uppercase
same with sd_port?

**Review by jjjoycing123 — APPROVED** [review]

**Review by jjjoycing123 — DISMISSED** [review]

File: securedrop/server.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+        await self.write_status(stream, msg)
+
+
+class ServerDriver:
+    def __init__(self, port=None, filename=None):
+        port = port if port is not None else sd_port
+        filename = filename if filename is not None else sd_filename
+        self.port, self.filename = port, filename
+        self.shm = shared_memory.SharedMemory(create=True, size=1)
+        self.shm.buf[0] = 0
```
Comment (BobbyBytes) [inline securedrop/server.py]: I think .shm is a little cryptic..

File: securedrop/server.py [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+            server.run(self.port, self.shm_name())
+        except Exception:
+            print("Caught exception. Exiting...")
+        finally:
+            self.shm.buf[0] = 1
+
+    def close(self):
+        self.shm.close()
+        self.shm.unlink()
+
```
Comment (BobbyBytes) [inline securedrop/server.py]: same as above

File: securedrop/tests/client_server_base_test.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+import time
+from contextlib import contextmanager
+import unittest
+import os
+from multiprocessing import Process, shared_memory
+
+from tornado.testing import AsyncTestCase
+
+from securedrop.client_server_base import ClientBase, ServerBase
```
Comment (BobbyBytes) [inline securedrop/tests/client_server_base_test.py:11]: Extra empty lines?

File: securedrop/tests/client_server_base_test.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+hostname = "localhost"
+port = 6969
+
+
+class EchoClient(ClientBase):
+    def __init__(self, data, resp, start_index=0, end_index=None):
+        super().__init__(hostname, port)
+        self.data = data
+        self.resp = resp
```
Comment (BobbyBytes) [inline securedrop/tests/client_server_base_test.py]: resp is cryptic to me.. what does it mean?
a stack pointer?

File: securedrop/tests/client_server_test.py [hunk 7]
```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python3
+
+import unittest
+from unittest.mock import patch
+
```
Comment (BobbyBytes) [inline securedrop/tests/client_server_test.py:5]: Extra line?

**Review by BobbyBytes — COMMENTED** [review]
Couple of things.. I think..

**Comment by aokellermann** [comment]
@BobbyBytes I'm going to wait for #24 to address the formatting comments

File: bin/securedrop_server [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+import os
+
+from securedrop import server
+
+
+if __name__ == "__main__":
+    home_dir = os.getenv("HOME")
+    if home_dir is None:
+        raise RuntimeError("HOME not found")
+
```
Comment (BobbyBytes) [inline bin/securedrop_server:12]: Extra empty line here?

**Review by BobbyBytes — DISMISSED** [review]
In addition to the tests here I manually  tested both executables and was able to successfully register and add contacts.

File: bin/securedrop_server (hunk 8 above)
Comment (aokellermann) [reply bin/securedrop_server:12]: Waiting on #24

**Comment by jjjoycing123** [comment]
I am having trouble registering users. I have deleted the json files but am still having issues. I will throw a screen shot up on Discord.

**Pull request merged**